### PR TITLE
Electron Selector Update

### PR DESCRIPTION
**Pull request by jdandoy**
This is a pretty big update to ElectronSelector to address #737 .  It'd be great if @mmilesi @fscutti  and @kratsg could take a look before it's added.  There may be a few more things to do, but this should be a start.

### CODE REVIEW

**Comment by johnda102** [comment]
Looks fine to me

File: Root/ElectronSelector.cxx [hunk 1]
```diff
@@ -278,15 +276,15 @@ EL::StatusCode ElectronSelector :: initialize ()
 
   if ( // m_LHOperatingPoint != "VeryLoose"       &&
        m_LHOperatingPoint != "Loose"           &&
-       m_LHOperatingPoint != "LooseAndBLayer"  &&
+       m_LHOperatingPoint != "LooseBL"  &&
```
Comment (mmilesi) [inline Root/ElectronSelector.cxx]: Any specific reason for changing the name of the WP?

File: Root/ElectronSelector.cxx [hunk 2]
```diff
@@ -96,29 +96,27 @@ ElectronSelector :: ElectronSelector (std::string className) :
...
-  m_confDirPID              = "mc15_20150224";
 
   // likelihood-based PID
+  m_doLHPID                 = true;
   m_doLHPIDcut              = false;
   m_LHOperatingPoint        = "Loose";
-  m_LHConfigYear            = "2015";
 
   // cut-based PID
+  m_doCutBasedPID           = true;
```
Comment (mmilesi) [inline Root/ElectronSelector.cxx]: Shall we set this to "false" here?

**Comment by mmilesi** [comment]
Looks good to me as well (I have left few comments for you to address)

File: Root/ElectronSelector.cxx (hunk 1 above)
Comment (jdandoy) [inline Root/ElectronSelector.cxx]: LooseBL is the name of the [working point ](https://twiki.cern.ch/twiki/bin/viewauth/AtlasProtected/EGammaIdentificationRun2#Usage_of_the_Electron_ID_tool)in AsgElectronLikelihoodTool.  I guess they changed it, and it's unfortunate that it's different from the electron efficiency naming convention, but we should stick with the tool naming convention.

**Comment by fscutti** [comment]
all good to me as well

File: Root/ElectronSelector.cxx (hunk 1 above)
Comment (mmilesi) [inline Root/ElectronSelector.cxx]: I see. But if possible I would stick to the old "LooseAndBLayer" naming, as it is consistent w/ the naming of the electron SFs. You can always let the code internally make the conversion to the naming that works for the tool.
Do you agree?

Comment (jdandoy) [inline Root/ElectronSelector.cxx]: I generally prefer for xAODAnaHelpers to not assume too much and to stick as close to the CP tools as possible.  But it's a bad situation and also breaks backwards compatibility, so I'll change it to accept both LooseBL and LooseAndBLayer.

**Comment by mmilesi** [comment]
All good from my side, thanks!

**Pull request merged**